query/physicalplan/filter.go

```diff
@@ -197,6 +243,237 @@ func (a *OrExpr) String() string {
 	return "(" + a.Left.String() + " OR " + a.Right.String() + ")"
 }
 
+type ArithmeticExpr struct {
+	operation   string
+	left        *ArrayRef
+	leftScalar  scalar.Scalar
+	right       *ArrayRef
+	rightScalar scalar.Scalar
+}
+
+func (m *ArithmeticExpr) Eval(r arrow.Record) (*Bitmap, error) {
+	return nil, fmt.Errorf("not used")
+}
+
+func (m *ArithmeticExpr) Project(mem memory.Allocator, r arrow.Record) ([]arrow.Field, []arrow.Array, error) {
```

Comment:
Just a reminder that you probably need to handle nulls here otherwise we'll get weird results and/or panics.